kernel/arch/dreamcast/kernel/init.c

```diff
@@ -280,7 +281,17 @@ void arch_shutdown() {
 }
 
 /* Generic kernel exit point (configurable) */
-void arch_exit() {
+__noreturn void arch_exit() {
+    exit(0);
```

Comment:
Maybe make a note that we're explicitly returning 0 here, which signifies a successful shutdown, because once return values matter, this is not the way you exit with an error code. This call should really just be used for graceful exit (or never used directly again).